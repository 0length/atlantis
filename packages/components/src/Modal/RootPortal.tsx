import { ReactNode } from "react";
import ReactDOM from "react-dom";

const modalRoot = document.body.appendChild(document.createElement("div"));

interface RootPortalProps {
  children: ReactNode | ReactNode[];
}

export function RootPortal({ children }: RootPortalProps) {
  return ReactDOM.createPortal(children, modalRoot);
}
