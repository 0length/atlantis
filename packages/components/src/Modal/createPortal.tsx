import React, { ReactNode, useEffect } from "react";
import ReactDOM from "react-dom";

interface CreatePortalProps {
  children: ReactNode | ReactNode[];
}

export function CreatePortal({ children }: CreatePortalProps) {
  return <Portal>{children}</Portal>;
}

interface PortalProps {
  children: ReactNode | ReactNode[];
}

export function Portal({ children }: PortalProps) {
  const modalRoot = document.body.appendChild(document.createElement("div"));
  const tempVar = document.createElement("div");

  useEffect(() => {
    modalRoot.appendChild(tempVar);

    return () => {
      modalRoot.removeChild(tempVar);
    };
  });

  return ReactDOM.createPortal(children, modalRoot);
}
