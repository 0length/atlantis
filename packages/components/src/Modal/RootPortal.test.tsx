import React from "react";
import { cleanup, render } from "@testing-library/react";
import { RootPortal } from "./RootPortal";
import { Modal } from "./Modal";

afterEach(cleanup);

test("Portal componet should render modal", () => {
  const title = "imma title";
  const content = "imma content";
  const handleClose = jest.fn();

  const modal = (
    <Modal title={title} open={true} onRequestClose={handleClose}>
      {content}
    </Modal>
  );

  const portal = render(<RootPortal>{modal}</RootPortal>);
  expect(portal).toMatchSnapshot();
});
