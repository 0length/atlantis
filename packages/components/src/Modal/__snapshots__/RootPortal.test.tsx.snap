// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal componet should render modal 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="overlay"
          style="opacity: 0;"
        />
        <div
          class="modal"
          style="opacity: 0.8; transform: scale(0.8) translateZ(0);"
        >
          <div
            class="header"
            data-testid="modal-header"
          >
            <h3
              class="base extraBold large uppercase"
            >
              imma title
            </h3>
            <button
              aria-label="Close modal"
              class="closeButton"
            >
              <svg
                class="icon base"
                viewBox="0 0 1024 1024"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class=""
                  d="M512 451.669l-225.835-225.835c-8.047-7.772-18.825-12.073-30.012-11.976s-21.888 4.585-29.799 12.495c-7.911 7.911-12.398 18.612-12.495 29.799s4.204 21.964 11.976 30.012l225.835 225.835-225.835 225.835c-7.772 8.047-12.073 18.825-11.976 30.012s4.585 21.888 12.495 29.798c7.91 7.91 18.612 12.399 29.799 12.497s21.965-4.203 30.012-11.977l225.835-225.835 225.835 225.835c8.047 7.774 18.825 12.075 30.012 11.977s21.888-4.587 29.798-12.497c7.91-7.91 12.399-18.611 12.497-29.798 0.094-11.187-4.203-21.965-11.977-30.012l-225.835-225.835 225.835-225.835c4.075-3.936 7.326-8.644 9.562-13.85s3.413-10.804 3.46-16.469c0.051-5.665-1.028-11.284-3.174-16.527s-5.312-10.007-9.318-14.013c-4.006-4.006-8.772-7.174-14.016-9.319-5.239-2.146-10.859-3.225-16.525-3.176s-11.264 1.226-16.469 3.462c-5.205 2.236-9.916 5.487-13.85 9.562l-225.835 225.835z"
                />
              </svg>
            </button>
          </div>
          imma content
        </div>
      </div>
    </div>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
